### PR TITLE
Apr 159 add provider to roatp

### PR DESCRIPTION
**Pull request by David-B-Read**
Dependency on das-roatp-service PR:
https://github.com/SkillsFundingAgency/das-roatp-service/pull/20

Note to reviewers: this story contains add RoATP organisation functionality with mandatory field validation only, the rest of the validation functionality will be implemented in story APR-233

### CODE REVIEW

File: src/SFA.DAS.AssessorService.Web.Staff/Controllers/Roatp/AddRoatpOrganisationController.cs [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        }
+
+        [Route("enter-details")]
+        public async Task<IActionResult> AddOrganisationDetails(AddOrganisationViewModel model)
+        {
+            var validationMessages = _validator.ValidateProviderType(model.ProviderTypeId);
+            if (validationMessages.Any())
+            {
+                model.ValidationErrors = new List<string>();
+                model.ValidationErrors.AddRange(validationMessages);
```
Comment (MarkFCain) [inline src/SFA.DAS.AssessorService.Web.Staff/Controllers/Roatp/AddRoatpOrganisationController.cs]: these two lines could be 1 line?

File: src/SFA.DAS.AssessorService.Web.Staff/Controllers/Roatp/AddRoatpOrganisationController.cs [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        public async Task<IActionResult> AddOrganisationPreview(AddOrganisationViewModel model)
+        {
+            model.OrganisationTypes = await _apiClient.GetOrganisationTypes(model.ProviderTypeId);
+            model.ProviderTypes = await _apiClient.GetProviderTypes();
+
+            var validationMessages = _validator.ValidateOrganisationDetails(model);
+            if (validationMessages.Any())
+            {
+                model.ValidationErrors = new List<string>();
+                model.ValidationErrors.AddRange(validationMessages);
```
Comment (MarkFCain) [inline src/SFA.DAS.AssessorService.Web.Staff/Controllers/Roatp/AddRoatpOrganisationController.cs]: 2 lines can be one?

File: src/SFA.DAS.AssessorService.Web.Staff/Controllers/Roatp/AddRoatpOrganisationController.cs [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+            var success = await _apiClient.CreateOrganisation(request);
+
+            if (!success)
+            {
+                model.OrganisationTypes = await _apiClient.GetOrganisationTypes(model.ProviderTypeId);
+                model.ProviderTypes = await _apiClient.GetProviderTypes();
+                model.ValidationErrors = new List<string>
+                {
+                    $"An error occurred when adding the organisation '{model.LegalName}'.<br/> Please try again later."
```
Comment (MarkFCain) [inline src/SFA.DAS.AssessorService.Web.Staff/Controllers/Roatp/AddRoatpOrganisationController.cs]: We could do with these being in a central resx maybe? That more for tech debt, so don't resolve here.

File: src/SFA.DAS.AssessorService.Web.Staff/Controllers/Roatp/AddRoatpOrganisationController.cs [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+                TradingName = model.TradingName,
+                OrganisationData = new OrganisationData
+                {
+                    CharityNumber = model.CharityNumber,
+                    CompanyNumber = model.CompanyNumber,
+                    FinancialTrackRecord = true,
+                    NonLevyContract = false,
+                    ParentCompanyGuarantee = false
+                },
+                UKPRN = Convert.ToInt64(model.UKPRN),
```
Comment (MarkFCain) [inline src/SFA.DAS.AssessorService.Web.Staff/Controllers/Roatp/AddRoatpOrganisationController.cs]: This looks like it needs a TryParse

Comment (David-B-Read) [reply src/SFA.DAS.AssessorService.Web.Staff/Controllers/Roatp/AddRoatpOrganisationController.cs]: Validation will be covered off in another story, APR-233 - anything non numeric or too small/large will be rejected before we get to here

File: src/SFA.DAS.AssessorService.Web.Staff/Validators/Roatp/AddOrganisationValidationErrorMessages.cs [hunk 5]
```diff
@@ -0,0 +1,11 @@
+﻿namespace SFA.DAS.AssessorService.Web.Staff.Validators.Roatp
+{
+    public static class AddOrganisationValidationErrorMessages
+    {
+        public static string InvalidProviderTypeId = "Tell us what the new training provider is";
```
Comment (MarkFCain) [inline src/SFA.DAS.AssessorService.Web.Staff/Validators/Roatp/AddOrganisationValidationErrorMessages.cs]: needs a tech debt story to get these words in a resx maybe

Comment (David-B-Read) [reply src/SFA.DAS.AssessorService.Web.Staff/Validators/Roatp/AddOrganisationValidationErrorMessages.cs]: Can do this when we play the rest of the validation messages

File: src/SFA.DAS.AssessorService.Web.Staff/Controllers/Roatp/AddRoatpOrganisationController.cs (hunk 3 above)
Comment (David-B-Read) [reply src/SFA.DAS.AssessorService.Web.Staff/Controllers/Roatp/AddRoatpOrganisationController.cs]: Will be picked up in APR-233

**Pull request merged**